File: Samples-Xamarin/Fragments/Dataform/DataFormValidationBehaviorFragment.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Util;
using Android.Views;
using Android.Widget;
using Android.Support.V4.View;

using Com.Telerik.Widget.Dataform.Engine;
using Com.Telerik.Widget.Dataform.Visualization;
using Com.Telerik.Widget.Dataform.Visualization.Core;

namespace Samples
{
	public class DataFormValidationBehaviorFragment : Android.Support.V4.App.Fragment, ExampleFragment
	{
		public String Title() {
			return "Validation Behavior";
		}

		public override View OnCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
			ViewGroup rootLayout = (ViewGroup)inflater.Inflate(Resource.Layout.fragment_dataform_grouping, null);

			RadDataForm dataForm = new RadDataForm(this.Activity);
			dataForm.LayoutManager = new DataFormLinearLayoutManager(this.Activity);
			Person person = new Person();
			person.Name = "Joe";
			dataForm.SetEntity(person);

			EntityPropertyEditor nameEditor = Android.Runtime.Extensions.JavaCast<EntityPropertyEditor>(dataForm.GetExistingEditorForProperty("Name"));
			nameEditor.Property().Validator = new NonEmptyValidator();
			nameEditor.ValidationViewBehavior = new ValidationAnimationBehavior(this.Activity);

			EntityPropertyEditor mailEditor = Android.Runtime.Extensions.JavaCast<EntityPropertyEditor>(dataForm.GetExistingEditorForProperty("Mail"));
			mailEditor.ValidationViewBehavior = new BlinkValidationBehavior(this.Activity);

			rootLayout.AddView(dataForm);

			return rootLayout;
		}

		public class BlinkValidationBehavior : DataFormValidationViewBehavior, Java.Lang.IRunnable {
			private View editorView;
			private bool stage1 = true;
			private bool stage2 = true;
			private bool stage3 = true;

			public BlinkValidationBehavior(Context context) : base(context) {
			}

			protected override void ShowNegativeFeedback(ValidationInfo info) {
				base.ShowNegativeFeedback(info);

				this.editorView = this.Editor.RootLayout();

				ViewCompat.Animate(editorView).Alpha(0).SetDuration(50).WithEndAction(this);
			}

			public void Run() {
				if (stage1) {
					ViewCompat.Animate(editorView).Alpha(1).SetDuration(50).WithEndAction(this);
					stage1 = false;
					return;
				}

				if (stage2) {
					ViewCompat.Animate (editorView).Alpha (0).SetDuration (50).WithEndAction (this);
					stage2 = false;
					return;
				}

				if (stage3) {
					ViewCompat.Animate (editorView).Alpha (1).SetDuration (50);
				}

				stage1 = true;
				stage2 = true;
			}
		}
	}
}

File: Samples-Java/Samples/src/main/java/fragments/dataform/DataFormValidationBehaviorFragment.java
package fragments.dataform;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.telerik.android.sdk.R;
import com.telerik.widget.dataform.engine.NonEmptyValidator;
import com.telerik.widget.dataform.engine.ValidationInfo;
import com.telerik.widget.dataform.visualization.DataFormLinearLayoutManager;
import com.telerik.widget.dataform.visualization.DataFormValidationViewBehavior;
import com.telerik.widget.dataform.visualization.RadDataForm;
import com.telerik.widget.dataform.visualization.ValidationAnimationBehavior;
import com.telerik.widget.dataform.visualization.core.EntityPropertyEditor;

import activities.ExampleFragment;

public class DataFormValidationBehaviorFragment extends Fragment implements ExampleFragment {
    @Override
    public String title() {
        return "Validation Behavior";
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        ViewGroup rootLayout = (ViewGroup)inflater.inflate(R.layout.fragment_dataform_grouping, null);

        RadDataForm dataForm = new RadDataForm(this.getActivity());
        dataForm.setLayoutManager(new DataFormLinearLayoutManager(this.getActivity()));
        Person person = new Person();
        person.setName("Joe");
        dataForm.setEntity(person);

        EntityPropertyEditor nameEditor = (EntityPropertyEditor) dataForm.getExistingEditorForProperty("Name");
        nameEditor.property().setValidator(new NonEmptyValidator());
        nameEditor.setValidationViewBehavior(new ValidationAnimationBehavior(this.getActivity()));

        EntityPropertyEditor mailEditor = (EntityPropertyEditor) dataForm.getExistingEditorForProperty("Mail");
        mailEditor.setValidationViewBehavior(new BlinkValidationBehavior(this.getActivity()));

        rootLayout.addView(dataForm);

        return rootLayout;
    }

    public class BlinkValidationBehavior extends DataFormValidationViewBehavior {
        public BlinkValidationBehavior(Context context) {
            super(context);
        }

        @Override
        protected void showNegativeFeedback(ValidationInfo info) {
            super.showNegativeFeedback(info);

            final View editorView = this.editor.rootLayout();

            ViewCompat.animate(editorView).alpha(0).setDuration(50).withEndAction(new Runnable() {
                @Override
                public void run() {
                    ViewCompat.animate(editorView).alpha(1).setDuration(50).withEndAction(new Runnable() {
                        @Override
                        public void run() {
                            ViewCompat.animate(editorView).alpha(0).setDuration(50).withEndAction(new Runnable() {
                                @Override
                                public void run() {
                                    ViewCompat.animate(editorView).alpha(1).setDuration(50);
                                }
                            });
                        }
                    });
                }
            });
        }
    }
}
